cmake/scripts/common/Platform.cmake

```diff
@@ -1,6 +1,13 @@
 if(NOT CORE_SYSTEM_NAME)
   string(TOLOWER ${CMAKE_SYSTEM_NAME} CORE_SYSTEM_NAME)
 endif()
+# Switch used path, if CORE_SOURCE_DIR is set use it (e.g. on addons build)
+# otherwise use the present source dir
+if(CORE_SOURCE_DIR)
+  set(PLATFORM_USED_SOURCE_DIR ${CORE_SOURCE_DIR})
+else()
+  set(PLATFORM_USED_SOURCE_DIR ${CMAKE_SOURCE_DIR})
+endif()
```

Comment:
Please explain this block. Both kodi and addons have CORE_SOURCE_DIR defined and pointing to the same main kodi source root, so we should be able to use CORE_SOURCE_DIR directly

Reply:
Is a bit a hen and egg problem.

On Kodi build it must be included on begin and `CORE_SOURCE_DIR` not set:
https://github.com/xbmc/xbmc/blob/master/CMakeLists.txt#L32
This then becomes set by https://github.com/xbmc/xbmc/blob/master/CMakeLists.txt#L36, and I think https://github.com/xbmc/xbmc/blob/master/cmake/scripts/common/Macros.cmake#L8 need a bit the defines from Platform.cmake.

Reply:
from a quick look it shouldn't be a problem to set CORE_SOURCE_DIR on top of the main CMakeLists.txt

Reply:
Have checked it does not work in other way without a bigger rewrite.

- `Platform.cmake` define the `CORE_SYSTEM_NAME`
- `CORE_SYSTEM_NAME` required by `Macros.cmake`
- `CORE_SOURCE_DIR` on `common/Macros.cmake` not allowed to set before, otherwise `${CORE_SYSTEM_NAME}/Macros.cmake` is not included (where need `CORE_SYSTEM_NAME`)

Not sure if this request fix something on my screensavers.rsxs Jenkins problem, but maybe it bring something :-)

Reply:
@wsnipex what you think about?

Reply:
I still think this should be refactored, but since that needs deeper changes we can leave it as is for v18. But please add a TODO comment that this should be fixed for v19

Reply:
Also heard that v19 is planning to restructure the CMake for addons. Maybe we could integrate it with it (or maybe it's already ;-)). Otherwise, I will see what can be done.